bitnami-labs/sealed-secrets
# Prometheus metrics & mixin

Rebased Prometheus PR on master.

Closes: #177 

- add http handler on `/metrics`. 
  There aren't any sensitive endpoints on this port, but if metrics are considered sensitive for some then I can move to a custom port.
- add standard `build_info` and `go_build_info` metrics
- add custom unseal error & request metrics - RE(D)
- add mixin with:
  - Grafana dashboard
  - Alerts
- add `PodMonitor` jsonnet for those using Prometheus Operator (us)

Additional thoughts:
- `make test` is currently failing for me on master as well as the `prometheus` branch: 
  `--- FAIL: TestHttpCert (1.37s)    `
- I did the bulk of this before key rotation was done and didn't experience any errors due to prometheus integration. 
- Key rotation metrics/dashboard/alert could be easily added in a subsequent PR.

Testing done:
- with `kind`
- create sealed secret - observe request counter increment
- submit SS with new secret - observe request counter increment
- submit SS with modified namespace (incorrect) - observe error increment
- submit SS with modified encrypted value (incorrect) - observe error increment
- delete secret - observe request counter increment
- delete SS - observe request counter increment
